### PR TITLE
Inline docstrings added for WC, NetCore, InProcess signer and txn executor solutions 669

### LINKED ISSUE #669: Write Docsstring for InProcessSigner, InProcessTransactionExecutor, NetCore and WalletConnect

### PR DESCRIPTION
**Pull request by robGG1997**
Closes #669

### CODE REVIEW

**Review by sneakzttv — APPROVED** [review]
yay function info!

File: src/ChainSafe.Gaming.InProcessTransactionExecutor/InProcessTransactionExecutor.cs [hunk 1]
```diff
@@ -13,12 +13,22 @@
 
 namespace ChainSafe.Gaming.InProcessTransactionExecutor
 {
+    /// <summary>
+    /// Concrete Implementation of <see cref="ITransactionExecutor"/> using In Process.
```
Comment (kalambet) [inline src/ChainSafe.Gaming.InProcessTransactionExecutor/InProcessTransactionExecutor.cs:17]: Same here. Pretty odd 'In Process' usage. Maybe re-phrase it some how.

File: src/ChainSafe.Gaming.WalletConnect/Models/TransactionModel.cs [hunk 2]
```diff
@@ -2,26 +2,50 @@
...
         public string From { get; set; }
 
+        /// <summary>
+        /// Transaction Receiver's public address.
+        /// </summary>
         [JsonProperty("to")]
         public string To { get; set; }
 
+        /// <summary>
+        /// Gas/Fee for transaction in Hex Value.
```
Comment (kalambet) [inline src/ChainSafe.Gaming.WalletConnect/Models/TransactionModel.cs:23]:
```suggestion
        /// Gas fee for transaction in Hex Value.
```

File: src/ChainSafe.Gaming.WalletConnect/WCLogWriter.cs [hunk 3]
```diff
@@ -5,19 +5,38 @@
...
+        /// Initializes a new instance of the <see cref="WCLogWriter"/> class.
+        /// </summary>
+        /// <param name="logWriter">Log Writer used for logging.</param>
         public WCLogWriter(ILogWriter logWriter)
         {
             this.logWriter = logWriter;
         }
 
+        /// <summary>
+        /// Log Message .
```
Comment (kalambet) [inline src/ChainSafe.Gaming.WalletConnect/WCLogWriter.cs:25]:
```suggestion
        /// Log Message.
```

File: src/ChainSafe.Gaming.WalletConnect/WalletConnectConfig.cs [hunk 4]
```diff
@@ -8,56 +8,122 @@
 
 namespace ChainSafe.Gaming.WalletConnect
 {
+    /// <summary>
+    /// Used for configuring a wallet connect session.
+    /// </summary>
     [Serializable]
-    [JsonObject(MemberSerialization.OptIn)]
+    [JsonObject(MemberSerialization.OptIn)] // Serialize fields explicitly or only ones with [JsonProperty] attribute.
```
Comment (kalambet) [inline src/ChainSafe.Gaming.WalletConnect/WalletConnectConfig.cs:15]: Can you please move to the `<remarks>` filed of the docstring

File: src/ChainSafe.Gaming.InProcessSigner/InProcessSignerExtensions.cs [hunk 5]
```diff
@@ -5,22 +5,42 @@
...
+        /// <returns>The same service collection that was passed in. This enables fluent style.</returns>
         public static IWeb3ServiceCollection UseInProcessSigner(this IWeb3ServiceCollection collection)
         {
             collection.AssertServiceNotBound<ISigner>();
             collection.AddSingleton<ISigner, InProcessSigner>();
             return collection;
         }
 
+        /// <summary>
+        /// Bind a <see cref="configuration"/> to <see cref="collection"/>.
```
[the remark this answers is not in the document]
Comment (robGG1997) [reply src/ChainSafe.Gaming.InProcessSigner/InProcessSignerExtensions.cs:39]: I'm sorry I don't follow, _case_ as in lower/upper case?

File: src/ChainSafe.Gaming.InProcessSigner/InProcessSigner.cs [hunk 6]
```diff
@@ -10,23 +10,47 @@
 
 namespace ChainSafe.Gaming.InProcessSigner
 {
+    /// <summary>
+    /// Concrete Implementation of <see cref="ISigner"/> using In Process.
```
[the remark this answers is not in the document]
Comment (robGG1997) [reply src/ChainSafe.Gaming.InProcessSigner/InProcessSigner.cs]: I'm sorry 😅 I had no idea what to write here, but I've updated it based on my understanding.

File: src/ChainSafe.Gaming.InProcessSigner/InProcessSigner.cs [hunk 7]
```diff
@@ -10,23 +10,47 @@
...
+    /// </summary>
     public class InProcessSigner : ISigner
     {
         private EthECKey privateKey;
         private EthereumMessageSigner messageSigner;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InProcessSigner"/> class.
+        /// </summary>
+        /// <param name="config">Injected Config for In Process.</param>
```
[the remark this answers is not in the document]
Comment (robGG1997) [reply src/ChainSafe.Gaming.InProcessSigner/InProcessSigner.cs]: Yep, updated.

File: src/ChainSafe.Gaming.InProcessTransactionExecutor/InProcessTransactionExecutor.cs (hunk 1 above)
Comment (robGG1997) [reply src/ChainSafe.Gaming.InProcessTransactionExecutor/InProcessTransactionExecutor.cs]: Yep, updated.

File: src/ChainSafe.Gaming.WalletConnect/IWalletConnectCustomProvider.cs [hunk 8]
```diff
@@ -2,12 +2,15 @@
...
+    /// <summary>
+    /// Connect and disconnect to a wallet via Wallet Connect and make a Json RPC request.
+    /// </summary>
     public interface IWalletConnectCustomProvider
     {
         /// <summary>
         /// connects using Wallet Connect.
         /// </summary>
-        /// <returns>connect address.</returns>
+        /// <returns>connected address.</returns>
```
[the remark this answers is not in the document]
Comment (robGG1997) [reply src/ChainSafe.Gaming.WalletConnect/IWalletConnectCustomProvider.cs]: Updated.

File: src/ChainSafe.Gaming.WalletConnect/Models/ChainModel.cs [hunk 9]
```diff
@@ -1,9 +1,21 @@
 namespace ChainSafe.Gaming.WalletConnect.Models
 {
+    /// <summary>
+    /// Chain model containing fields used for Wallet Connect.
+    /// </summary>
     public class ChainModel
     {
+        /// <summary>
+        /// default namespace for EVM.
```
[the remark this answers is not in the document]
Comment (robGG1997) [reply src/ChainSafe.Gaming.WalletConnect/Models/ChainModel.cs]: Updated.

File: src/ChainSafe.Gaming.WalletConnect/Models/TransactionModel.cs (hunk 2 above)
Comment (robGG1997) [reply src/ChainSafe.Gaming.WalletConnect/Models/TransactionModel.cs]: Updated.

File: src/ChainSafe.Gaming.WalletConnect/WCLogWriter.cs (hunk 3 above)
Comment (robGG1997) [reply src/ChainSafe.Gaming.WalletConnect/WCLogWriter.cs]: Updated.

File: src/ChainSafe.Gaming.WalletConnect/WalletConnectConfig.cs (hunk 4 above)
Comment (robGG1997) [reply src/ChainSafe.Gaming.WalletConnect/WalletConnectConfig.cs]: Yeah that makes more sense, updated.

File: src/ChainSafe.Gaming.InProcessSigner/InProcessSignerExtensions.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (kalambet) [reply src/ChainSafe.Gaming.InProcessSigner/InProcessSignerExtensions.cs]: So i was not sure if:
```
<see cref="configuration"/> to <see cref="collection"/>
```
or 
```
<see cref="Configuration"/> to <see cref="Collection"/>
```
matters and if those types even exist.

[the remark this answers is not in the document]
Comment (robGG1997) [reply src/ChainSafe.Gaming.InProcessSigner/InProcessSignerExtensions.cs]: these aren't references to types, they're references to the method's parameters.

**Review by kalambet — APPROVED** [review]
Great!

**Review by kantagara — APPROVED** [review]

**Pull request merged**